Requirements/HasPermissionRequirement.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.AspNetCore.Authorization;
+
+using GeekHub.Models;
+
+namespace GeekHub.Requirements
+{
+    public class HasPermissionRequirement: IAuthorizationRequirement
+    {
+        public Permission Permission { get; }
+
+        public HasPermissionRequirement(Permission permission)
+        {
+            Permission = permission;
```

Comment:
Is this meant to be a predicate or a settor?

Reply:
What do you mean?

Reply:
Is this function supposed to return a boolean or just set the permission? If it's supposed to set, i think we should rename it to `SetPermissionRequirement`.